kwave/utils/checks.py

```diff
@@ -143,35 +117,37 @@ def is_unix():
     Check whether the current platform is a Unix-like system.
 
     Args:
-        None
 
     Returns:
-        bool: True if the current platform is a Unix-like system, False otherwise.
+        is_unix: (bool) True if the current platform is a Unix-like system, False otherwise.
 
     """
     return platform.system() in ['Linux', 'Darwin']
 
 
-def check_stability(kgrid, medium):
+def check_stability(kgrid: "kWaveGrid", medium: "kWaveMedium") -> float:
     """
-          checkStability calculates the maximum time step for which the k-space
-          propagation models kspaceFirstOrder1D, kspaceFirstOrder2D and
-          kspaceFirstOrder3D are stable. These models are unconditionally
-          stable when the reference sound speed is equal to or greater than the
-          maximum sound speed in the medium and there is no absorption.
-          However, when the reference sound speed is less than the maximum
-          sound speed the model is only stable for sufficiently small time
-          steps. The criterion is more stringent (the time step is smaller) in
-          the absorbing case.
-
-          The time steps given are accurate when the medium properties are
-          homogeneous. For a heterogeneous media they give a useful, but not
-          exact, estimate.
+    Calculates the maximum time step for which the k-space
+    propagation models are stable.
+
+    These models are unconditionally
+    stable when the reference sound speed is equal to or greater than the
+    maximum sound speed in the medium and there is no absorption.
+    However, when the reference sound speed is less than the maximum
+    sound speed the model is only stable for sufficiently small time
+    steps. The criterion is more stringent (the time step is smaller) in
+    the absorbing case.
+
+    The time steps given are accurate when the medium properties are
+    homogeneous. For a heterogeneous media they give a useful, but not
+    exact, estimate.
+
     Args:
-        kgrid: k-Wave grid object return by kWaveGrid
-        medium: structure containing the medium properties
+        kgrid: (kWaveGrid) k-Wave grid object return by kWaveGrid
```

Comment:
* `return by kWaveGrid` => seems redundant to me
* `(kWaveGrid)` => do we need to specify type in the docstring given that it is already part of the function signature?

Reply:
to part one you are right....

part two.... [you are also right](https://stackoverflow.com/questions/65768215/type-annotations-with-google-style-docstrings)!  I didn't realize this 😮